service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/CreateAzureNetworkInterfaceStepUnitTest.java

```diff
@@ -0,0 +1,182 @@
+package bio.terra.workspace.service.resource.controlled.cloud.azure.vm;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.workspace.amalgam.landingzone.azure.LandingZoneApiDispatch;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.common.exception.AzureManagementExceptionUtils;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.management.exception.ManagementError;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.compute.ComputeManager;
+import com.azure.resourcemanager.network.NetworkManager;
+import com.azure.resourcemanager.network.models.NetworkInterfaces;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.verify;
+
+@Tag("unit")
+@ExtendWith(MockitoExtension.class)
+public class CreateAzureNetworkInterfaceStepUnitTest {
+
+  @Mock AzureConfiguration azureConfiguration;
+  @Mock CrlService crlService;
+  @Mock ControlledAzureVmResource resource;
+  @Mock SamService samService;
+  @Mock WorkspaceService mockWorkspaceService;
+  @Mock ResourceDao resourceDao;
+  @Mock LandingZoneApiDispatch landingZoneApiDispatch;
+  @Mock NetworkManager networkManager;
+
+
+
+  @BeforeEach
+  void setup() {
```

Comment:
Nit: delete the empty setup.

Perhaps this changes in a later commit… I'm trying to review commit by commit to avoid re-reviewing the whole PR.